Formula/dnslookup.rb

```diff
@@ -0,0 +1,20 @@
+class Dnslookup < Formula
+  desc "Simple command-line utility to make DNS lookups using any protocol"
+  homepage "https://github.com/ameshkov/dnslookup/"
+  url "https://github.com/ameshkov/dnslookup.git",
+      tag:      "v1.4.1",
+      revision: "99af6421f9c7f409a2c645d81bd3d8212414296d"
+  license "GPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags",
+             "-X main.VersionString=#{version}",
+             *std_go_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/dnslookup -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.